crates/binstalk/src/ops/resolve.rs

```diff
@@ -91,44 +91,63 @@ async fn resolve_inner(
         .collect::<Result<Vec<_>, _>>()?;
     let resolvers = &opts.resolvers;
 
-    let mut handles: Vec<(Arc<dyn Fetcher>, _)> =
-        Vec::with_capacity(desired_targets.len() * resolvers.len());
+    let mut binary_name = None;
+    if package_info.binaries.len() == 1 {
```

Comment:
I think you could do a match here

```suggestion
    match package_info.binaries.as_slice() {
```

Reply:
sure